File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_313_Data_Structures/ExCode/20-Set-Map/LeetCode/LeetCode-0001-2.java
class Solution {
    /// O(n) algorithm with O(n) extra space
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> myMap = new HashMap();

        myMap.put(nums[0], 0);
        for (int i=1; i<nums.length; i++){
            Integer val = myMap.get(target-nums[i]);
            if (val != null)
                return new int[]{val, i};

            myMap.put(nums[i], i);  // Insert this number to the map
        } //end-for

        return new int[]{-1, -1};
    } //end-twoSum
}

File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_313_Data_Structures/ExCode/20-Set-Map/LeetCode/LeetCode-0001-2.cpp
class Solution {
public:
    // O(n) algorithm with O(1) extra space
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> myMap;

        for (int i=0; i<nums.size(); i++){
            int otherNum = target-nums[i];
            if (myMap.count(otherNum))
                return {myMap[otherNum], i};
            else
                myMap[nums[i]] = i;
        } // end-for
        return {-1, -1};
    } // end-twoSum
};